setup.py

```diff
@@ -35,7 +35,43 @@
     "ipykernel>=6.29.0",
 ]
 
-rag = ["sentence_transformers", "pypdf", "ipython", "beautifulsoup4", "markdownify"]
+retrieve_chat = ["chromadb", "sentence_transformers", "pypdf", "ipython", "beautifulsoup4", "markdownify"]
```

Comment:
"chromadb" is not supposed to be in the list. Maybe there's a reason and I missed it.

Reply:
chromadb is a hard dependency for all RAG agent types because Chroma is coupled tightly with RAG agent type. Once we remove chromaDB as a hard dependency, we can remove it from base retrieve chat requirements. Until then, when qdrant or pgvector is installed, it also needs chromadb

Reply:
As for the open ai version, when on the older version, mypy throws an error that it is unable to find openAI vector_stores which is only available in latest (1.23.3)+

Reply:
Thanks. We can keep the chromadb. For OpenAI version, we need to figure out how to support older versions.

Reply:
Should that be scoped for this PR? Or does that need to be addressed as a separate issue and another PR? I'm not sure how to approach the backwards compatibility requirement.

Reply:
@IANTHEREAL is it true that openai<1.23.3 support is broken by #2466 ? Can we fix it easily? I agree that the fix is out of the scope of this PR.

Reply:
> @IANTHEREAL is it true that openai<1.23.3 support is broken by #2466 ? Can we fix it easily? I agree that the fix is out of the scope of this PR. Is it OK to merge this PR and then fix the problem? Or is it better to fix that problem first?

@sonichi in the PR #2466 , the previous versions <1.21 has been removed. maybe adding them back would fix the problem. i would prefer, fix that problem and then merge this PR.

Reply:
Thanks @Hk669 . @Knucklessg1 For a good user experience of backward compatibility, can we find a different solution to the mypy check error instead of increasing the required openai version? Could you share which lines fail the mypy check so that we can take a look how to resolve it?

Reply:
maybe we can remove openai utils from the mypy check list for now and add it back when the compatibility issue fixed. @sonichi

Reply:
That works.

Reply:
I must have forgotten to add #type: ignore; that should work.


![image](https://github.com/microsoft/autogen/assets/10701973/102e868d-4809-4c22-9459-83e459f2d977)

Reply:
@WaelKarkoub kindly offers addressing the remaining issues in #2490 
So merging the PR now.